serde_arrow/src/internal/schema/from_samples/mod.rs

```diff
@@ -694,4 +702,131 @@ mod test {
             expected,
         );
     }
+
+    #[test]
+    fn example_enum_as_union() {
+        let expected = json!({
+          "name": "$",
+          "data_type": "Union",
+          "children": [
+            {
+              "name": "Real",
+              "data_type": "Struct",
+              "children": [
+                {
+                  "name": "value",
+                  "data_type": "F32",
+                }
+              ]
+            },
+            {
+              "name": "Complex",
+              "data_type": "Struct",
+              "children": [
+                {
+                  "name": "i",
+                  "data_type": "F32"
+                },
+                {
+                  "name": "j",
+                  "data_type": "F32"
+                }
+              ]
+            }
+          ]
+        });
+
+        test_to_tracer(
+            &[
+                Number::Real { value: 1.0 },
+                Number::Complex { i: 0.5, j: 0.5 },
+            ],
+            TracingOptions::default(),
+            expected,
+        );
+    }
+
+    #[test]
+    fn example_enum_as_struct() {
```

Comment:
should probably combine this test and the following one to check for equality between the generated fields using both the enum and the struct